packages/angular_devkit/schematics/src/tree/host-tree.ts

```diff
@@ -180,7 +180,7 @@ export class HostTree implements Tree {
         case 'c': {
           const { path, content } = action;
 
-          if ((this._willCreate(path) || this._willOverwrite(path))) {
+          if ((this._willCreate(path) || this._willOverwrite(path) || (this.exists(path) && overwriteConflictAllowed))) {
```

Comment:
I don't think the `overwriteConflichAllowed` check is needed here.
Looking at the failing E2E test (https://github.com/angular/angular-cli/blob/e7e01d213e9fb2473dfec4ec531e53f4c5822fe6/tests/legacy-cli/e2e/tests/generate/component/component-duplicate.ts), it appears that it only passed because it relied on the defect that is being fixed in this PR.  The test is creating the same exact component twice so the second should be a no-op (and not fail since it doesn't do anything).  To make the test work, something would need to be changed in one of the component's files after the first generate command so that there would be a legitimate conflict.

Reply:
Good call, I reverted the update to the exists logic and updated that test to update the contents of the app component ts file.